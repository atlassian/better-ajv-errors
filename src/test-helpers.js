import { readFileSync } from 'fs';
import { getFixturePath } from 'jest-fixtures';
import Ajv from 'ajv';

export async function getSchemaAndData(name, dirPath) {
  const schemaPath = await getFixturePath(dirPath, name, 'schema.json');
  const schema = JSON.parse(readFileSync(schemaPath, 'utf8'));
  const dataPath = await getFixturePath(dirPath, name, 'data.json');
  const data = JSON.parse(readFileSync(dataPath, 'utf8'));

  return [schema, data];
}

export function evaluateSchema(schema, data) {
  const ajv = new Ajv();
  const fn = ajv.compile(schema);
  fn(data);
  expect(fn.errors.length).toBeGreaterThanOrEqual(0);
  return fn.errors;
}
