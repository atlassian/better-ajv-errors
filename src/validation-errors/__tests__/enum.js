import { getSchemaAndData } from '../../test-helpers';
import EnumValidationError from '../enum';

describe('Enum', () => {
  let schema, data;
  beforeAll(async () => {
    [schema, data] = await getSchemaAndData('enum', __dirname);
  });

  it('prints correctly for enum prop', () => {
    const error = new EnumValidationError({
      keyword: 'enum',
      dataPath: '/id',
      schemaPath: '#/enum',
      params: { allowedValues: ['foo', 'bar'] },
      message: `should be equal to one of the allowed values`,
    });

    expect(error.print(schema, data)).toMatchSnapshot();
  });

  it('prints correctly for no levenshtein match', () => {
    const error = new EnumValidationError({
      keyword: 'enum',
      dataPath: '/id',
      schemaPath: '#/enum',
      params: { allowedValues: ['one', 'two'] },
      message: `should be equal to one of the allowed values`,
    });

    expect(error.print(schema, data)).toMatchSnapshot();
  });

  it('prints correctly for empty value', () => {
    const error = new EnumValidationError({
      keyword: 'enum',
      dataPath: '/id',
      schemaPath: '#/enum',
      params: { allowedValues: ['foo', 'bar'] },
      message: `should be equal to one of the allowed values`,
    });

    expect(error.print(schema, { id: '' })).toMatchSnapshot();
  });
});
