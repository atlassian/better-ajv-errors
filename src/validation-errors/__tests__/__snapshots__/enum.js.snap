// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Enum prints correctly for empty value 1`] = `
Array [
  "[31m[1mENUM[22m[39m[31m should be equal to one of the allowed values[39m",
  "[31m(foo, bar)[39m
",
  "[90m1│ {[39m",
  "[31m2│   \\"id\\": \\"\\"[39m",
  "[36m 〉[39m",
  "[36m 〉   ☝🏽  Unexpected value, should be equal to one of the allowed values[39m",
  "[36m 〉[39m",
  "[90m3│ }[39m",
]
`;

exports[`Enum prints correctly for enum prop 1`] = `
Array [
  "[31m[1mENUM[22m[39m[31m should be equal to one of the allowed values[39m",
  "[31m(foo, bar)[39m
",
  "[90m1│ {[39m",
  "[31m2│   \\"id\\": \\"baz\\"[39m",
  "[36m 〉[39m",
  "[36m 〉   ☝🏽  Did you mean [1mbar[22m here?[39m",
  "[36m 〉[39m",
  "[90m3│ }[39m",
]
`;

exports[`Enum prints correctly for no levenshtein match 1`] = `
Array [
  "[31m[1mENUM[22m[39m[31m should be equal to one of the allowed values[39m",
  "[31m(one, two)[39m
",
  "[90m1│ {[39m",
  "[31m2│   \\"id\\": \\"baz\\"[39m",
  "[36m 〉[39m",
  "[36m 〉   ☝🏽  Unexpected value, should be equal to one of the allowed values[39m",
  "[36m 〉[39m",
  "[90m3│ }[39m",
]
`;
